File: src/main/java/org/blacksun/view/NetworkPanel.java
package org.blacksun.view;

import org.blacksun.network.Network;
import org.blacksun.utils.Config;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;
import java.awt.*;

public class NetworkPanel extends JScrollPane {
    private final Network network;
    private final NetworkRender render;

    public NetworkPanel(@NotNull Network network) {
        super();
        this.network = network;
        render = new NetworkRender(network);
        viewport.setView(render);
        Config cfg = Config.getConfig();
        setPreferredSize(new Dimension(cfg.getInt("viewW"),
                cfg.getInt("viewH")));
    }

    public void update() {
        render.update();
        revalidate();
    }

    public Network getNetwork() {
        return network;
    }
}

File: src/main/java/org/blacksun/view/NetworkPanel.kt
package org.blacksun.view

import org.blacksun.network.Network
import org.blacksun.utils.Config

import javax.swing.*
import java.awt.*

class NetworkPanel(val network: Network) : JScrollPane() {
    private val render: NetworkRender = NetworkRender(network)

    init {
        viewport.view = render
        val cfg = Config.config
        preferredSize = Dimension(cfg.getInt("viewW"),
                cfg.getInt("viewH"))
    }

    fun update() {
        render.update()
        revalidate()
    }
}
